libethereum/EthereumHost.cpp

```diff
@@ -512,27 +512,25 @@ tuple<vector<shared_ptr<EthereumPeer>>, vector<shared_ptr<EthereumPeer>>, vector
     vector<shared_ptr<EthereumPeer>> chosen;
     vector<shared_ptr<EthereumPeer>> allowed;
     vector<shared_ptr<SessionFace>> sessions;
-
-    size_t peerCount = 0;
+    double percentDecimal = _percent / 100.0;
     foreachPeer([&](std::shared_ptr<EthereumPeer> _p)
     {
         if (_allow(_p.get()))
         {
-            allowed.push_back(_p);
             sessions.push_back(_p->session());
+            // Random selection of peers according to percent requested
+            if(((double)rand() / RAND_MAX) < percentDecimal)
```

Comment:
Please use C++ random number tools from `<random>` instead of C stuff.

(I recommend this video to quit the bad habit of `rand()` https://channel9.msdn.com/Events/GoingNative/2013/rand-Considered-Harmful)

Reply:
You could also use `std::random_shuffle`

Reply:
Is it assumed that allowed's elements are in a certain order? Or can I just return a shuffled allowed vector?

Reply:
The goal is to randomly select them and order doesn't matter